openstudiocore/src/model/AirflowNetworkHorizontalOpening.cpp

```diff
@@ -0,0 +1,234 @@
+  }
+
+  bool AirflowNetworkHorizontalOpening_Impl::isAirMassFlowExponentWhenOpeningisClosedDefaulted() const {
+    return isEmpty(OS_AirflowNetworkHorizontalOpeningFields::AirMassFlowExponentWhenOpeningisClosed);
+  }
+
+  double AirflowNetworkHorizontalOpening_Impl::slopingPlaneAngle() const {
+    boost::optional<double> value = getDouble(OS_AirflowNetworkHorizontalOpeningFields::SlopingPlaneAngle,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool AirflowNetworkHorizontalOpening_Impl::isSlopingPlaneAngleDefaulted() const {
+    return isEmpty(OS_AirflowNetworkHorizontalOpeningFields::SlopingPlaneAngle);
+  }
+
+  double AirflowNetworkHorizontalOpening_Impl::dischargeCoefficient() const {
+    boost::optional<double> value = getDouble(OS_AirflowNetworkHorizontalOpeningFields::DischargeCoefficient,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool AirflowNetworkHorizontalOpening_Impl::setAirMassFlowCoefficientWhenOpeningisClosed(double airMassFlowCoefficientWhenOpeningisClosed) {
+    bool result = setDouble(OS_AirflowNetworkHorizontalOpeningFields::AirMassFlowCoefficientWhenOpeningisClosed, airMassFlowCoefficientWhenOpeningisClosed);
+    return result;
+  }
+
+  bool AirflowNetworkHorizontalOpening_Impl::setAirMassFlowExponentWhenOpeningisClosed(double airMassFlowExponentWhenOpeningisClosed) {
+    bool result = setDouble(OS_AirflowNetworkHorizontalOpeningFields::AirMassFlowExponentWhenOpeningisClosed, airMassFlowExponentWhenOpeningisClosed);
+    return result;
+  }
+
+  void AirflowNetworkHorizontalOpening_Impl::resetAirMassFlowExponentWhenOpeningisClosed() {
+    bool result = setString(OS_AirflowNetworkHorizontalOpeningFields::AirMassFlowExponentWhenOpeningisClosed, "");
+    OS_ASSERT(result);
+  }
+
+  bool AirflowNetworkHorizontalOpening_Impl::setSlopingPlaneAngle(double slopingPlaneAngle) {
+    bool result = setDouble(OS_AirflowNetworkHorizontalOpeningFields::SlopingPlaneAngle, slopingPlaneAngle);
+    return result;
+  }
+
+  void AirflowNetworkHorizontalOpening_Impl::resetSlopingPlaneAngle() {
+    bool result = setString(OS_AirflowNetworkHorizontalOpeningFields::SlopingPlaneAngle, "");
+    OS_ASSERT(result);
+  }
+
+  bool AirflowNetworkHorizontalOpening_Impl::setDischargeCoefficient(double dischargeCoefficient) {
+    bool result = setDouble(OS_AirflowNetworkHorizontalOpeningFields::DischargeCoefficient, dischargeCoefficient);
+    return result;
+  }
+
+} // detail
+
+AirflowNetworkHorizontalOpening::AirflowNetworkHorizontalOpening(const Model& model,
+  double massFlowCoefficientWhenOpeningisClosed,
+  double dischargeCoefficient)
+  : ModelObject(AirflowNetworkHorizontalOpening::iddObjectType(), model)
+{
+  OS_ASSERT(getImpl<detail::AirflowNetworkHorizontalOpening_Impl>());
+
+  // TODO: Appropriately handle the following required object-list fields.
+  //OS_ASSERT(setHandle());
+  OS_ASSERT(setAirMassFlowCoefficientWhenOpeningisClosed(massFlowCoefficientWhenOpeningisClosed));
```

Comment:
@jasondegraw you don't want to do this type of initialization in an assert, the asserts are completely compiled out in release mode

Reply:
Oops, I thought I caught all of those. Sorry about that.